core/app/models/spree/order/checkout.rb

```diff
@@ -1,324 +1,338 @@
 
-              if states[:delivery]
-                before_transition to: :delivery, do: :create_proposed_shipments
-                before_transition to: :delivery, do: :ensure_available_shipping_rates
-                before_transition to: :delivery, do: :set_shipments_cost
-                before_transition from: :delivery, do: :apply_free_shipping_promotions
-              end
+            if states[:delivery]
+              before_transition to: :delivery, do: :create_proposed_shipments
+              before_transition to: :delivery, do: :ensure_available_shipping_rates
+              before_transition to: :delivery, do: :set_shipments_cost
+              before_transition from: :delivery, do: :apply_free_shipping_promotions
+            end
 
-              before_transition to: :resumed, do: :ensure_line_item_variants_are_not_deleted
-              before_transition to: :resumed, do: :ensure_line_items_are_in_stock
+            before_transition to: :resumed, do: :ensure_line_item_variants_are_not_deleted
+            before_transition to: :resumed, do: :ensure_line_items_are_in_stock
 
-              before_transition to: :complete, do: :ensure_line_item_variants_are_not_deleted
-              before_transition to: :complete, do: :ensure_line_items_are_in_stock
+            before_transition to: :complete, do: :ensure_line_item_variants_are_not_deleted
+            before_transition to: :complete, do: :ensure_line_items_are_in_stock
 
-              after_transition to: :complete, do: :finalize!
-              after_transition to: :resumed,  do: :after_resume
-              after_transition to: :canceled, do: :after_cancel
+            after_transition to: :complete, do: :finalize!
+            after_transition to: :resumed,  do: :after_resume
+            after_transition to: :canceled, do: :after_cancel
 
-              after_transition from: any - :cart, to: any - [:confirm, :complete] do |order|
-                order.update_totals
-                order.persist_totals
-              end
+            after_transition from: any - :cart, to: any - [:confirm, :complete] do |order|
+              order.update_totals
+              order.persist_totals
             end
+          end
 
-            alias_method :save_state, :save
+          alias_method :save_state, :save
+        end
+
+        def self.go_to_state(name, options = {})
+          self.checkout_steps[name] = options
+          previous_states.each do |state|
+            add_transition({ from: state, to: name }.merge(options))
+          end
+          if options[:if]
+            previous_states << name
+          else
+            self.previous_states = [name]
           end
+        end
 
-          def self.go_to_state(name, options={})
-            self.checkout_steps[name] = options
-            previous_states.each do |state|
-              add_transition({from: state, to: name}.merge(options))
+        def self.insert_checkout_step(name, options = {})
```

Comment:
Assignment Branch Condition size for insert_checkout_step is too high. [19.34/15]